File: testdata/bench/cpp/BenchmarkTest00007.cpp
#include <string>
#include <sqlite3.h>

void find_user(const std::string &login) {
    std::string query = "SELECT * FROM users WHERE login = '" + login + "'";
    sqlite3 *db;
    sqlite3_open("app.db", &db);
    sqlite3_exec(db, query.c_str(), nullptr, nullptr, nullptr);
    sqlite3_close(db);
}

File: testdata/bench/csharp/BenchmarkTest00007.cs
using System.Data.SqlClient;
using Microsoft.AspNetCore.Mvc;

namespace BenchApp.Controllers
{
    public class SqliController : Controller
    {
        [HttpGet]
        public IActionResult Login()
        {
            string user = Request.Query["username"];
            string pass = Request.Query["password"];
            var conn = new SqlConnection("Server=localhost;Database=app;");
            conn.Open();
            string sql = "SELECT * FROM users WHERE username = '" + user + "' AND password = '" + pass + "'";
            var cmd = new SqlCommand(sql, conn);
            var reader = cmd.ExecuteReader();
            return Ok();
        }
    }
}
